### ISSUE TITLE
Yii 3.0 compatibility

### ISSUE THREAD

**Issue by trntv** [yii2-starter-kit/yii2-starter-kit#651]

**Comment by XzAeRo** [comment]
Just in case. Yii 3+ is just a restructuring of the framework into smaller packages, so, in theory, it shouldn't take to much time to migrate everything.

Docs about Yii3 status and future roadmap for Yii: https://github.com/yiisoft/yii-core/blob/master/ROADMAP.md

**Comment by XzAeRo** [comment]
The new Yii3 approach will discontinue the "advanced app" template since they are adopting the "each application a service" approach. You can check more about this in [the UPGRADE.md file](https://github.com/yiisoft/yii-core/blob/master/UPGRADE.md) in the new [yii-core](https://github.com/yiisoft/yii-core/) project.

And why the advanced app will be discontinued? Because configuration files will have to be managed through composer: https://github.com/hiqdev/composer-config-plugin killing the multi-config approach we have in the advanced app and in this project. 

This is an extract of the UPGRADE.md file mentioned above.

![screenshot_2019-01-31 yiisoft yii-core 1](https://user-images.githubusercontent.com/1656037/52060136-5a0b2c00-254a-11e9-8318-396c30065f22.png)

So what does this mean? For the next upgrade we may need to separate each part (API, storage, frontend, backend, console), into it's own project, **OR** we try to make just one big simple application with several modules for each section.

What should we do?

**Comment by kwazaro** [comment]
I think one application with modules for "console", "api", "backend" etc. is the best way to go.

**Comment by Buliwyfa** [comment]
yes my opinion goes for 1 app with modules as @kwazaro  proposed

**Comment by XzAeRo** [comment]
Some updates for those who are interested:

- The new Yii3 application structures is vastly different from what we currently have in Yii2. There are no "modules", rather than different applications bundled together with a special URL router (similar to what Symfony/Django does). We now have RouterFactory, Middleware, Dispatchers, Entities and several other changes to the framework that make it impossible to migrate directly.
- The new Yii3 starter kit will be in a separate repository. We will try to match the features 1:1.
- The Yii2 starter kit will be maintained and updated until it's EOL.
- Given the new approach of Yii3, we will be creating starter kits with some popular frontend frameworks too (like React, Vue, etc.), besides the native renderer. I'm not really sure how to handle this, but the idea will be to have something like different repositories for each frontend.
